.github/workflows/test.yml

```diff
@@ -18,6 +18,6 @@ jobs:
         uses: beckermr/provision-with-micromamba@main
 
       - name: run python
-        shell: bash -l {0}
+        shell: bash -l -euo pipefail {0}
```

Comment:
@beckermr the action fails if `-u` option is added to bash, because it implement stricter tests on variables (fails if it can't find a variable instead of producing a null value)

Reply:
So I am running tests in another PR and github actions appears to be having trouble with unbound variables from the system defaults and not the changes to the action locally. See here: https://github.com/mamba-org/provision-with-micromamba/pull/2/checks?check_run_id=1474313370#step:5:4

Reply:
Ahhhhh. The shell init fails before the environment is activated. There is not a lot we can do about this I think.

Reply:
The quetz ci bash environment is probably not being fully setup since the bashrc init is ending early.

Reply:
yes, it's a tricky one. perhaps the action could somehow keep the original bash initialisation scripts, and only append to them?

Reply:
See my PR that does this. Even if that happens, the action still won't work because the bash init will fail on missing variables in the system scripts before it initializes the conda environment. 

https://github.com/mamba-org/provision-with-micromamba/pull/2

Reply:
thanks, this looks good, but still I don't understand, why it fails when using the action whereas it works when I install micromamba manually run exactly the same steps: https://github.com/btel/quetz/actions/runs/391596195/workflow#L36-L78

Reply:
The action calls `micromamba init test` in the `.bashrc`. When you add `-u` to the shell line, the shell initialization process never gets to the `micromamba init` call. Instead, the shell initialization process ends early. In your quetz CI scripts, you are calling `micromamba init` by hand afterwards. I would recommend removing `-u` from the quetz CI line and instead using `set -euo pipefail` directly in the run section.

Reply:
See this workflow run here: https://github.com/mamba-org/provision-with-micromamba/runs/1474564047?check_suite_focus=true

The workflow file has a step

```
      - name: test shell init
        shell: bash -l -euo pipefail {0}
        run: |
          echo 'shell is not fully setup!'
```

it outputs

```
Run echo 'shell is not fully setup!'
  echo 'shell is not fully setup!'
  shell: /bin/bash -l -euo pipefail {0}
/etc/profile.d/Z97-byobu.sh: line 24: LC_BYOBU: unbound variable
shell is not fully setup!
```

Notice that the bash init fails on a system script `/etc/profile.d/Z97-byobu.sh`. It never reaches any of the micromamba additions to the `.bashrc`.

Reply:
ah, now I understand. This could be also fixed by exporting `export LC_BYOBU=0` before activating mamba, unless there are more missing variables further down the script. Thanks a lot for help!